testing/supporting_docs/ui_selectors_and_their_importance.md

```diff
@@ -0,0 +1,178 @@
+resulting inefficient and more brittle xpaths, as produced by tools such as “Appium 
+Inspector”, being used and effecting test suite performance and reliability. 
+
+## Examples of adding Adding Selectors to Applications
+
+### iOS (AccessibilityId Example)
+
+By default, if an ‘accessibilityId’ is not defined for an element, an iOS application will 
+dynamically set the ‘accessibilitityId’ of an element to the value of the element. This is not 
+ideal when testing for any elements that display information that will be dynamically created, 
+such as a username for a user, or possibly with elements such as toggle switches or checkboxes 
+where the value is updated after being clicked.
+
+For iOS the ‘accessibilityId’ can be added to a UI component in two ways. When viewing the 
+storyboards in Xcode, to add an ‘accessibilityId’, firstly select the element, then open the 
+identity inspector tab. Under the ‘User Defined Runtime Attributes’, add a new Key Path called 
+‘accessibilityIdentifier’, set the Type to ‘String’, and set the Value to the desired name of 
+the new selector.
+
+
+![img text](./images/adding_accessibilityid_xcode_identity_inspector_panel.png "Adding AccessibilityId Via Xcode Identity Inspector Panel")
+
+The second way of adding an ‘accessibilityId’ to a UI element is to do so through Swift code.  
+This can be done by setting the ‘accessibilityIdentifier’ attribute of a UI element as shown 
+below.
+
+```swift
+if (indexPath.row == 0) {
+    fieldNameLabel.text = "Name"
+    fieldValueLabel.text = self.currentUser!.fullName
+    fieldValueLabel.accessibilityIdentifier = "username"
+} else {
+    fieldName.text = "Email"
+    fieldValueLabel.text = self.currentUser!.email
+    fieldValueLabel.accessibilityIdentifier = "email"
+}
+```
+
+### Android (AccessibilityId Example)
+Unlike iOS Android does not dynamically assign any values to the AccesibilityId.  Instead if 
+the developer wishes to use AccessibilityId for testing purposes, it can be added to Android 
+apps by setting the ‘contentDescription’ attribute either in the xml UI specifications or 
+through the use of code by using the ‘setContentDescription’ method of a UI element.  
+
+
+```xml
+<Button 
+    android:id="@+id/keycloakLogin"
+    android:contentDescription="@+string/authenticate_button"
+    style="@style/widget.AppCompat.Button.Colored"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:text="Authenticate"
+    android:layout_marginBottom="15dp"
+    android:layout_alignParentBottom="true"
+    android:layout_alignParentStart="true" />
+```
+
+
+![alt text](./images/authenticate_button_in_appium_inspector.png "Accessibility Id Now Available to Use")
+
+### Xamarin (AutomationId Example)
+Xamarin elements have a unique attribute called AutomationId, which can be used to set UI selectors 
+across both iOS and Android platforms.  This can be set in either the XAML code or in the C# backend 
```

Comment:
@twnolan consistency should be persisted with the word 'android' across the whole document, e.g. starting with either lower or upper case.

Reply:
upper case is correct